temporal/fx.go

```diff
@@ -726,7 +718,7 @@ func loadClusterInformationFromStore(ctx context.Context, svc *config.Config, cl
 				logger.Info(fmt.Sprintf("Use rpc address %v for cluster %v.", newMetadata.RPCAddress, metadata.ClusterName))
 			}
 		}
-		svc.ClusterMetadata.ClusterInformation[metadata.ClusterName] = newMetadata
+		svc.ClusterMetadata.ClusterInformation[metadata.ClusterName] = *newMetadata
```

Comment:
Maybe--I'm a little worried that this will cause unexpected mutations from other parts of the code that reference it, though, so I don't want to include it in this PR